x/poe/contract/tgrade_validator_voting_integration_test.go

```diff
@@ -5,17 +5,21 @@ import (
 	"encoding/json"
 	"sort"
 	"testing"
+	"time"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 
+	"github.com/confio/tgrade/app"
 	"github.com/confio/tgrade/x/poe/contract"
 	"github.com/confio/tgrade/x/poe/types"
 )
 
-//go:embed tg4_stake.wasm
+//go:embed tgrade_validator_voting.wasm
```

Comment:
On second thought, it will still be a random contract. The only requirement is that it supports the `migrate` call. I can revert to stake for clarity.